src/api/app/models/staging/workflow.rb

```diff
@@ -50,6 +50,31 @@ def write_to_backend
     Backend::Api::Sources::Project.write_staging_workflow(project.name, User.current_login, render_xml)
   end
 
+  def self.load_groups
+    # as it is not expected that there are many groups (~30) we cache all of them. Otherwise use this instead:
+    # group_ids = Review.where(bs_request: BsRequest.where(staging_project: @staging_projects)).select('group_id').distinct
+    # Group.where(id: group_ids).each do |group|
+    #
+    # TODO: Refactor this code using to_h when updating to Ruby 2.6 (performance improvement)
+    #
+    groups_hash = {}
+    # FIXME: try to use cache without failure. Like: Rails.cache.fetch(Group.all.cache_key) do
```

Comment:
What do you mean with "without failure"? What is exactly the problem? :confused:

Reply:
Using `Rails.cache.fetch(Group.all.cache_key)` always returns an empty hash :disappointed:

I simplified the code to make it work for the review. We can refactor afterwards.

Reply:
@Ana06 we can work on this thoroughly after the meeting if you want.

Reply:
Not sure what you tried. Can you please try the following code? I think it should work :wink:

```
Rails.cache.fetch(Group.all.cache_key) do
    groups_hash = {}
    Group.find_each do |group|
      groups_hash[group.title] = group
    end
    groups_hash
end

Reply:
If that doesn't work I would check if `cache_key` is correctly generated (in other words if it is always the same

Reply:
`each`is not the problem. I'll check `cache_key`

Reply:
@saraycp, do you have groups in your instance? That would explain that it is {}

Reply:
I do :smile: